hack/release/common.sh

```diff
@@ -66,136 +53,165 @@ authenticatePrivateRepo() {
 }
 
-publishHelmChart() {
-    CHART_NAME=$1
-    RELEASE_VERSION=$2
-    RELEASE_REPO=$3
-    HELM_CHART_VERSION=$(helmChartVersion "$RELEASE_VERSION")
-    HELM_CHART_FILE_NAME="${CHART_NAME}-${HELM_CHART_VERSION}.tgz"
-    AH_CONFIG_FILE_NAME="${CHART_NAME}/artifacthub-repo.yaml"
-
-    cd charts
-    [[ -s "${AH_CONFIG_FILE_NAME}" ]] && oras push "${RELEASE_REPO}:artifacthub.io" --config /dev/null:application/vnd.cncf.artifacthub.config.v1+yaml "${AH_CONFIG_FILE_NAME}:application/vnd.cncf.artifacthub.repository-metadata.layer.v1.yaml"
-    helm dependency update "${CHART_NAME}"
-    helm lint "${CHART_NAME}"
-    helm package "${CHART_NAME}" --version "${HELM_CHART_VERSION}"
-    helm push "${HELM_CHART_FILE_NAME}" "oci://${RELEASE_REPO}"
-    rm "${HELM_CHART_FILE_NAME}"
-    cd ..
-
-    cosignHelmChart "${RELEASE_REPO}/${CHART_NAME}" "${HELM_CHART_VERSION}"
+buildDate() {
+  local date_epoch
+
+  date_epoch="${1}"
+
+  date -u -r "${date_epoch}" "+%Y-%m-%dT%H:%M:%SZ" 2>/dev/null
 }
 
-cosignHelmChart() {
-    RELEASE_REPO=$1
-    HELM_CHART_VERSION=$2
-    digest="$(crane digest "${RELEASE_REPO}:${HELM_CHART_VERSION}")"
-    cosign sign --yes "${RELEASE_REPO}:${HELM_CHART_VERSION}@${digest}"
+prepareWebsite() {
+  local version version_parts short_version
+
+  version="${1}"
+  # shellcheck disable=SC2206
+  version_parts=(${version//./ })
+  short_version="${version_parts[0]}.${version_parts[1]}"
+
+  createNewWebsiteDirectory "${short_version}"
+  removeOldWebsiteDirectories
+  editWebsiteConfig "${version}"
+  editWebsiteVersionsMenu
 }
 
 createNewWebsiteDirectory() {
-    RELEASE_VERSION=$1
-    versionData "${RELEASE_VERSION}"
+  local short_version="${1}"
 
-    mkdir -p "website/content/en/${RELEASE_MINOR_VERSION}"
-    cp -r website/content/en/preview/* "website/content/en/${RELEASE_MINOR_VERSION}/"
+  mkdir -p "website/content/en/v${short_version}"
+  cp -r website/content/en/preview/* "website/content/en/v${short_version}/"
 
-    # Update parameterized variables in the preview documentation to be statically set in the versioned documentation
-    find "website/content/en/${RELEASE_MINOR_VERSION}/" -type f | xargs perl -i -p -e "s/{{< param \"latest_release_version\" >}}/${RELEASE_VERSION}/g;"
-    find "website/content/en/${RELEASE_MINOR_VERSION}/" -type f | xargs perl -i -p -e "s/{{< param \"latest_k8s_version\" >}}/$(yq .params.latest_k8s_version website/hugo.yaml)/g;"
-    find website/content/en/${RELEASE_MINOR_VERSION}/*/*/*.yaml -type f | xargs perl -i -p -e "s/preview/${RELEASE_MINOR_VERSION}/g;"
-    find "website/content/en/${RELEASE_MINOR_VERSION}/" -type f | xargs perl -i -p -e "s/{{< githubRelRef >}}/\/${RELEASE_VERSION}\//g;"
+  # Update parameterized variables in the preview documentation to be statically set in the versioned documentation
+  find "website/content/en/v${short_version}/" -type f -print0 | xargs perl -i -p -e "s/{{< param \"latest_release_version\" >}}/${RELEASE_VERSION}/g;"
```

Comment:
I had recalled doing this and this messed with the output format in a way that `xargs` couldn't understand. I'll double-check as well, but you're sure that adding `-print0` here doesn't mess with the success of this command?

Reply:
Shellcheck is adamant that it's needed but I'll check again.

Reply:
Sure thing. Let me know. I had also gotten warnings for not adding this before but I recalled that adding it had caused issues

Reply:
You're absolutely right, so I've supressed the Shellcheck warnings so it's clear this is the desired outcome.